### PR TITLE
ASAP: Int2019 57 request funds reciever view

### PR DESCRIPTION
**Pull request by radulambrino**

### CODE REVIEW

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    private getDataForTransfer = () => {
+          return {
+              ...this.getTransferDataFromUI(),
+              isTransferRequested: this.state.isTransferRequested,
+              requestId: this.state.requestId
+          }
+    }
+
+    transferMoney = () => {
+
```
Comment (vasi-axinte) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: please remove this empty line. Also, the formatting is a bit weird for this method. Can you check it out?

File: Server/VikingVault.API/Controllers/TransferRequestsController.cs [hunk 2]
```diff
@@ -25,6 +25,35 @@ public TransferRequestsController(ITransferRequestService transferRequestService
...
+                return Ok("Couldn't send data to server!");
+            }
+            catch(Exception e)
+            {
+                if(e is DatabaseException)
+                {
+                    return NotFound(e.Message);
+                }
+
+                return Ok("Server Error."); //TO BE CHANGED
```
Comment (vasi-axinte) [inline Server/VikingVault.API/Controllers/TransferRequestsController.cs]: Please change it or remove the comment :)

**Review by vasi-axinte — CHANGES_REQUESTED** [review]

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+import './TransferFunds.css';
+import '../ExchangeForm.css';
+import TransferFundsModal from './TransferFundsModal';
+import Toggle from '../../Common/Toggle';
+import TransferRequests from '../TransferRequests';
+
+const transferUrl = `${constants.baseUrl}transferFunds`;
+const requestUrl = `${constants.baseUrl}transferRequests`;
+const currencyMap = {
+  [currencyEnum.ron]: 0,
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: This mapping is a bit rendundant as TypeScript has enum types...make use of them instead

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+class TransferFundsPage extends React.Component<any, ITransferFundsState>{
+    constructor(props:any){
+        super(props);
+        this.state = {
+          transferedAmount: 0,
+          currency: "EUR",
+          cardNumber: "Card Number",
+          transferDetails: "Trasnfer Details",
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: Typo

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx (hunk 1 above)
Comment (radulambrino) [reply ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: What do you mean by formatting?

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx (hunk 3 above)
Comment (radulambrino) [reply ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: We used this mapping across different parts of the code so I'd rather keep it consistent

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx (hunk 1 above)
Comment (radulambrino) [reply ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: Nevermind, solved it.

File: Server/VikingVault.API/Controllers/TransferRequestsController.cs [hunk 5]
```diff
@@ -28,6 +28,35 @@ public TransferRequestsController(ITransferRequestService transferRequestService
...
+                return Ok("Couldn't send data to server!");
+            }
+            catch(Exception e)
+            {
+                if(e is DatabaseException)
+                {
+                    return NotFound(e.Message);
+                }
+
+                return Ok("Server Error.");
```
Comment (Slitthe) [inline Server/VikingVault.API/Controllers/TransferRequestsController.cs:56]: You return the HTTP status code of `Ok` but the message is `"Server Error."

File: ClientApp/viking-vault/src/UserDashboard/TransferRequests.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                <thead>
+                    <tr>
+                        <th scope="col">From</th>
+                        <th scope="col">Amount</th>
+                        <th scope="col">Currency</th>
+                        <th scope="col">Details</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {this.state.transferRequests.map( (transferRequest) => {
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/TransferRequests.tsx:91]: Give this poor parameter (`transferRequest`) a type

**Pull request merged**